lib/Support/LoweringOptions.cpp

```diff
@@ -88,11 +88,21 @@ void LoweringOptions::parse(StringRef text, ErrorHandlerT errorHandler) {
       disallowPortDeclSharing = true;
     } else if (option == "printDebugInfo") {
       printDebugInfo = true;
+    } else if (option == "spillWiresAtPrepare") {
+      spillWiresAtPrepare = true;
     } else {
       errorHandler(llvm::Twine("unknown style option \'") + option + "\'");
       // We continue parsing options after a failure.
     }
   }
+
+  if (spillWiresAtPrepare && !disallowLocalVariables) {
```

Comment:
Does this need to be an error? I think it won't lead to invalid Verilog to use both, it's just that the spill into a local variable won't occur. Is that the case?

If so, I guess this is about protecting ourselves so that we get the spills we are expecting? For example, that spilling we do to work around the Verilator performance cliff. I guess this is safer in the short term, until #3437 is done.

Reply:
Yes, this should be short term. Actually I've seen errors that sv.wire is generated in procedural regions without `disallowLocalVariables`.